src/test/java/org/dependencytrack/parser/cyclonedx/CycloneDXExporterTest.java

```diff
@@ -0,0 +1,50 @@
+package org.dependencytrack.parser.cyclonedx;
+
+import org.cyclonedx.exception.GeneratorException;
+import org.dependencytrack.PersistenceCapableTest;
+import org.dependencytrack.model.Component;
+import org.dependencytrack.model.Severity;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.tasks.scanners.AnalyzerIdentity;
+
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+
+public class CycloneDXExporterTest extends PersistenceCapableTest {
+
+    @Test // https://github.com/DependencyTrack/dependency-track/issues/3834
+    public void testSchemaValidityOfExportedVexWithMultipleComponentsWithSameVulnerability() throws GeneratorException {
```

Comment:
Consider moving this to [`VexResourceTest`](https://github.com/DependencyTrack/dependency-track/blob/master/src/test/java/org/dependencytrack/resources/v1/VexResourceTest.java), and asserting the entire response JSON.

Also, a test case that demonstrates what happens when `componentA` and `componentB` have different analyses (i.e. one is suppressed the other is not, or one has sate `IN_TRIAGE` the other `EXPLOITABLE`) would be good.

If we de-duplicate vulnerabilities without the `Analysis` in mind, we might drop important data.

Reply:
Yeah I'm re-implementing it such that the whole Vulnerability object is used for comparison. Still in the process of making it work.

Will do what you wrote with the tests.

Reply:
Done